cyclops/report/plot/classification.py

```diff
@@ -93,6 +95,116 @@ def _set_class_names(self, class_names: List[str]) -> None:
             class_names = [f"Class_{i+1}" for i in range(self.class_num)]
         self.class_names = class_names
 
+    def calibration(
+        self,
+        data: pd.DataFrame,
+        y_true_col: str,
+        y_prob_col: str,
+        group_col: Optional[str] = None,
+        title: Optional[str] = "Calibration Plot",
+        layout: Optional[go.Layout] = None,
+        **plot_kwargs: Any,
+    ) -> go.Figure:
+        """Plot calibration curve for binary classification.
+
+        Parameters
+        ----------
+        data : pd.DataFrame
+            Dataframe containing true labels and predicted probabilities
+        y_true_col : str
+            Column name for true labels
+        y_prob_col : str
+            Column name for predicted probabilities
+        group_col : str, optional
+            Column name for grouping the data, by default None
+        title: str, optional
+            Plot title, by default "Calibration Plot"
+        layout : go.Layout, optional
+            Customized figure layout, by default None
+        **plot_kwargs : dict
+            Additional keyword arguments
+
+        Returns
+        -------
+        go.Figure
+            Plotly figure object
+
+        """
+        if self.task_type != "binary":
+            raise ValueError(
+                "Calibration plot is only available for binary classification"
+            )
+        # Create subplots: 1 plot for calibration curve, 1 plot for histogram
+        fig = make_subplots(
+            rows=2,
+            cols=1,
+            shared_xaxes=True,
+            vertical_spacing=0.02,
+            row_heights=[0.8, 0.2],
+        )
+        if group_col:
+            # Plot a calibration curve for each level of the grouping variable
+            unique_groups = data[group_col].unique()
+            for group in unique_groups:
+                group_df = data[data[group_col] == group]
+                prob_true, prob_pred = calibration_curve(
+                    group_df[y_true_col], group_df[y_prob_col], n_bins=10
```

Comment:
It would be great though if we could pass `n_bins` as an argument to the method.